openstack/neutron/templates/etc/_neutron-policy.json.tpl

```diff
@@ -7,13 +7,13 @@
     "admin": "role:network_admin and rule:owner",
     "context_is_network_admin": "rule:context_is_admin or rule:admin",
     "context_is_editor": "rule:context_is_network_admin or rule:member",
-    "context_is_viewer":  "rule:context_is_editor or rule:viewer",
+    "context_is_viewer":  "rule:context_is_editor or rule:viewer or rule:context_is_securitygroup_admin or rule:context_is_securitygroup_viewer",
```

Comment:
Doesn't that create a cyclic graph between `context_is_viewer` and `context_is_securitygroup_viewer`? Does that work?

Reply:
This should work. I think the "or" condition in context_is_securitygroup_viewer prevent issues here. I will try it out in qa and if it doesnt work i will roll it back.